networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/discovery/discv5/NodeRecordConverter.java

```diff
@@ -29,6 +29,9 @@
 
   private static DiscoveryPeer socketAddressToDiscoveryPeer(
       final NodeRecord nodeRecord, final InetSocketAddress address) {
-    return new DiscoveryPeer(((Bytes) nodeRecord.get(EnrField.PKEY_SECP256K1)), address);
+    return new DiscoveryPeer(
+        ((Bytes) nodeRecord.get(EnrField.PKEY_SECP256K1)),
+        address,
+        ((Bytes) nodeRecord.get("eth2")));
```

Comment:
Definitely going to want to make this `Optional` to handle the field not being set.

Reply:
Done.

Reply:
Done.